2018/Vinogradov/Tournament/Tournament/Messenger.cs

```diff
@@ -0,0 +1,240 @@
+            Console.WriteLine("Type \"exit\" to close the program.");
+            Console.WriteLine();
+
+            DrawTable(tournament);
+
+            Console.WriteLine("Enter your command:");
+        }
+
+        private static string CenterName(string name, int size)
+        {
+            StringBuilder refactored = new StringBuilder(size);
+            int firstGap = (size - name.Length) / 2;
+
+            for (int i = 0; i < firstGap; i++)
+            {
+                refactored.Append(' ');
+            }
+
+            refactored.Append(name);
+            int secondGap = size - name.Length - firstGap;
+
+            for (int i = 0; i < secondGap; i++)
+            {
+                refactored.Append(' ');
+            }
+
+            return refactored.ToString();
+        }
+
+        private static void DrawTable(TournamentController tournament)
+        {
+            string playerName;
+
+            for (int k = 0; k < tournament.Matches.Length; k++)
+            {
+                for (int i = 0; i < tournament.Matches[k].Length; i++)
+                {
+                    var currentMatch = tournament.Matches[k][i];
+
+                    for (int j = 0; j < 2; j++)
+                    {
+                        int playerNumber = currentMatch.Opponents[j];
+                        playerName = "";
+
+                        if (playerNumber >= 0)
+                        {
+                            playerName = tournament.Players[playerNumber];
+                        }
+
+                        if (currentMatch.Winner == j)
+                        {
+                            Console.ForegroundColor = ConsoleColor.Green;
+                        }
+
+                        Console.Write(CenterName(playerName, (maxChars + 2) * (int)Math.Pow(2, k)));
+                        Console.ResetColor();
+                    }
+
+                }
+
+                Console.WriteLine();
+
+                for (int i = 0; i < tournament.Matches[k].Length * 2; i++)
+                {
+                    Console.Write(CenterName("|", (maxChars + 2) * (int)Math.Pow(2, k)));
+                }
+
+                Console.WriteLine();
+
+                for (int i = 0; i < tournament.Matches[k].Length; i++)
+                {
+                    Console.Write(CenterName("|", (maxChars + 2) * (int)Math.Pow(2, k + 1)));
+                }
+
+                Console.WriteLine();
+            }
+
+            playerName = "";
+
+            if (tournament.Champion >= 0)
+            {
+                playerName = tournament.Players[tournament.Champion];
+            }
+
+            Console.ForegroundColor = ConsoleColor.Green;
+            Console.WriteLine(CenterName(playerName, (maxChars + 2) * (int)Math.Pow(2, tournament.Matches.Length)));
+            Console.ResetColor();
+        }
+    }
+}
```

Comment:
Maybe it would be better if the part with the execution wasn't in the Messenger. The Messenger should be working with console only and only displaying stuff, no logic like choice and execution should be here